apiserver/facades/controller/usersecretsdrain/register.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package usersecretsdrain
+
+import (
+	"reflect"
+
+	"github.com/juju/errors"
+
+	commonsecrets "github.com/juju/juju/apiserver/common/secrets"
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/secrets/provider"
+	"github.com/juju/juju/state"
+)
+
+// Register is called to expose a package of facades onto a given registry.
+func Register(registry facade.FacadeRegistry) {
+	registry.MustRegister("UserSecretsDrain", 1, func(ctx facade.Context) (facade.Facade, error) {
+		return newUserSecretsDrainAPI(ctx)
+	}, reflect.TypeOf((*SecretsDrainAPI)(nil)))
+}
+
+// newUserSecretsDrainAPI creates a SecretsDrainAPI for draining user secrets.
+func newUserSecretsDrainAPI(context facade.Context) (*SecretsDrainAPI, error) {
+	if !context.Auth().AuthController() {
+		return nil, apiservererrors.ErrPerm
+	}
+	leadershipChecker, err := context.LeadershipChecker()
```

Comment:
Do we need leadership checks for user secrets?

Reply:
No need for the user secrets stuff, but the shared code in the common package needs them.